### DISCUSSION TITLE
Integration of Paparazzi

### DISCUSSION THREAD

**Discussion by amrhus26** [paparazzi/paparazzi#3037] [Q&A] [answered]
Hello, I want to ask a question about integration of paparazzi with QGroundControl.
1. Can we use paparazzi to simulate multiple UAV controlled by 1 flight operator of QGroundControl ?
2. Can we inject in QGC other traffic coming from paparazzi ?

**Comment by Fabien-B** [comment, marked as answer]
Hello,
Some of the paparazzi concepts (e.g flight plans, ...) really differs from other autopilots such as pixhawk or ardupilot, so we made our own Ground Control Station. (There is a new one, see https://github.com/paparazzi/PprzGCS).
Paparazzi uses PprzLink to communicate, which has the same role as MavLink, but they are not compatible.

However there is a mavlink module that integrate some basic mavlink messages:
- https://github.com/paparazzi/paparazzi/blob/master/conf/modules/mavlink.xml
- https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/modules/datalink/mavlink.c#L116-L131
So I suppose that if you use that module, you should be able to see the paparazzi UAVs in QGC.
